typescript/cli/src/deploy/core.ts

```diff
@@ -461,3 +469,77 @@ async function writeAgentConfig(
   writeJson(outPath, agentConfig);
   logGreen('Agent configs written');
 }
+
+interface DeployParams {
+  context: WriteCommandContext;
+  chain: ChainName;
+  config: CoreConfig;
+}
+/**
+ * Executes the core deploy command.
+ */
+export async function runCoreDeploy({
+  context,
+  chain,
+  config,
+}: {
+  context: WriteCommandContext;
+  chain: ChainName;
+  config: CoreConfig;
+}) {
+  const {
+    signer,
+    isDryRun,
+    chainMetadata,
+    dryRunChain,
+    registry,
+    skipConfirmation,
+  } = context;
+
+  // Select a dry-run chain if it's not supplied
+  if (dryRunChain) {
+    chain = dryRunChain;
+  } else if (!chain) {
+    if (skipConfirmation) throw new Error('No chain provided');
+    chain = await runSingleChainSelectionStep(
+      chainMetadata,
+      'Select chain to connect:',
+    );
+  }
```

Comment:
I think we should have a way to pass a chain flag outside of dry run too

Reply:
as discussed, `chain` can be passed as a parameter